tests/integration/test_signals.py

```diff
@@ -0,0 +1,56 @@
+from django.test import override_settings
+
+import mock
+import pytest
+from katka import constants
+from katka.fields import username_on_model
+from katka.models import SCMPipelineRun, SCMStepRun
+
+
+@pytest.mark.django_db
+class TestSCMStepsRunSignals:
+    def test_steps_updated(self, scm_step_run, scm_pipeline_run):
+        assert scm_pipeline_run.steps_total == 1
+        assert scm_pipeline_run.steps_completed == 0
+        assert scm_pipeline_run.modified_username == 'initial'
+        assert scm_step_run.status == constants.STEP_STATUS_NOT_STARTED
+
+        scm_step_run.status = constants.STEP_STATUS_SUCCESS
+        with username_on_model(SCMStepRun, 'signal_tester'):
+            scm_step_run.save()
+
+        assert scm_pipeline_run.steps_total == 1
+        assert scm_pipeline_run.steps_completed == 1
+        assert scm_pipeline_run.modified_username == 'signal_tester'
+
+
+@pytest.mark.django_db
+class TestSCMPipelineRunSignals:
```

Comment:
Maybe add a test for when pipeline is in status INITIALIZING

Reply:
Yes, this is a good one, I'll add a test for this